lib/gateway/make-resolver.js

```diff
@@ -634,6 +634,15 @@ function makeResolver ({
         context
       })
 
+      if (service.collectors) {
+        collect({
```

Comment:
I will do the call only if there is at least one collector.

Reply:
I'm not sure about this because I'd have to iterate through the object values with something like `Object.values(service.collectors).some(field => field === true)` and then inside the check again inside the function `if (collectors.collectHeaders)`.
Usually if `service.collectors` is truthy it means that some collectors have been set to true so I'd rather check once.

Reply:
we can have 

```
collectors.collectHeaders = false
```

WE have to.collect 4 different type, at the end. There is no requirement to do a `find`. An if with an `||` is the fastest way to do it. 
Another way is to set the value once at the beginnning of the request and use it to check it.